go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/hashicorp/hcp
 
-go 1.22.2
+go 1.23
```

Comment:
@pierluc-codes should we stay pinned on Go v1.22.2? If so I will roll this back.